applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js

```diff
@@ -73,6 +78,19 @@ return view.extend({
 		o.value('dns', _('DNS'));
 		o.default = 'webroot';
 
+		if (!hasDnsApi) {
+			let opkgPackage = 'acme-acmesh-dnsapi';
+			o = s.taboption('general', form.Button, '_install');
+			o.depends('validation_method', 'dns');
+			o.title = _('Package is not installed');
+			o.inputtitle = _('Install package %s').format(opkgPackage);
+			o.inputstyle = 'apply';
+			o.onclick = function () {
+				window.open(L.url('admin/system/opkg') +
+					'?query=' + opkgPackage, '_blank', 'noopener');
+			};
```

Comment:
While this is neat, I feel like having an actual install button goes a bit against a separation of concerns principle. We are installing certificates here, we shouldn't be installing software; so I think it's better to just have a notice and let users go to the software install page themselves to install the missing package...

Reply:
Hmm - I think it's OK. In the p910nd case I built, there was so much butt-hurt trying to include the common print modules, so to avoid clicking and typing, I added buttons, since it's conceivable that many users need this combo which is not available out of the box - all users needed one or the other module, but nobody wanted both by default at the cost of an extra few kb. Anything we can do to make a smooth experience is worthwhile IMO.